### PR TITLE
Info2-lecture: Heapsort algorithm

### PR DESCRIPTION
**Pull request by Twhining**
Also added a directory for lecture related stuff

### CODE REVIEW

**Comment by C0DECYCLE** [comment]
Very nice thank you for you contribution! If its ok i will adjust the path so that it matches our patterns bit more.

**Comment by Twhining** [comment]
of course yea i wasn't sure how to do it.

File: Informatics II/Lectures/Twihning/heapsort.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <math.h>
+#define N 13
+
+int arr[N] = {0,2,7,8,3,9,4,1,6,11,14,12,17};
```
Comment (C0DECYCLE) [inline Informatics II/Lectures/Twihning/heapsort.c:5]: one minor thing, could you please move this array initialisation line inside the main.

**Review by C0DECYCLE — CHANGES_REQUESTED** [review]
Minor thing

File: Informatics II/Lectures/Twihning/heapsort.c [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    return i*2+2;
+}
+
+//determine the index of left child 
+int left(int i){
+    return i*2+1;
+}
+
+//convert the tree to a heap from the index i downwards
+void heapify(int *A, int i , int k){
```
Comment (Patertuck) [inline Informatics II/Lectures/Twihning/heapsort.c:24]: I would recommend to keep consistent variable names for things. This variable with the length of the array goes from being called "N" to "p" and "s" in Heap_sort into the "k" here. Just call it all "n" for example; It doesn't change the correctness of the code but is way easier to read and understand.

File: Informatics II/Lectures/Twihning/heapsort.c [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    int m, l, r;
+    m = i;
+    l = left(i);
+    r = right(i);
```
Comment (Patertuck) [inline Informatics II/Lectures/Twihning/heapsort.c:28]: This can be shortened as:
```suggestion
    int l = left(i);
    int r = right(i);
```

**Review by Patertuck — CHANGES_REQUESTED** [review]
Amazing code, love this. Just have some minor details that don't change the output for cleaner code. Thanks a lot for your contribution! <3

**Review by C0DECYCLE — APPROVED** [review]
LGTM!

**Comment by C0DECYCLE** [comment]
https://www.youtube.com/@CodeAesthetic

**Pull request merged**